pkg/nodeconfig/nodeconfig.go

```diff
@@ -346,6 +346,35 @@ func (nc *nodeConfig) configureCNI() error {
 	return nil
 }
 
+// Deconfigure removes the node from the cluster, reverting changes made by the Configure function
+func (nc *nodeConfig) Deconfigure() error {
+	drainHelper := &drain.Helper{Ctx: context.TODO(), Client: nc.k8sclientset}
+	// Set nc.node to the existing node
+	if err := nc.setNode(true); err != nil {
```

Comment:
I see, the source of the issue is that the `IP()` method of the windows interface should really be `Address()`, or `NetworkAddress()` and the struct should have `address` instead of `ipAddress`.

I can open a followup PR for these changes

Reply:
To be clear, I dont see a functionality issue, just a naming issue here.

Reply:
I see a functionality issue here, won't
```
if nc.IP() == nodeAddress.Address 
```
fail if only DNS is set?